### PR TITLE
Add info for vim in readme.

### PR DESCRIPTION
**Pull request by Carreau**

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -290,6 +290,27 @@ You can pass additional arguments to ``darker`` in the ``blackArgs`` option
 (e.g. ``["--diff", "--isort"]``), but make sure at least ``--diff`` is included.
 
 
+Vim
+---
+
+Unlike black and many other formatter darker needs access to the git history,
```
Comment (akaihola) [inline README.rst:296]:
```suggestion
Unlike black and many other formatters ``darker`` needs access to the Git history,
```

...just to unify spelling across the document.

File: README.rst [hunk 2]
```diff
@@ -290,6 +290,27 @@ You can pass additional arguments to ``darker`` in the ``blackArgs`` option
...
+therefore it does not work properly with the classical auto reformat plugins.
+
+You can though ask vim to run doarker on file save with the following in your
+`.vimrc`:
```
Comment (akaihola) [inline README.rst:300]:
```suggestion
``.vimrc``:
```

Double backticks required in RestructuredText to get monospace.

File: README.rst [hunk 3]
```diff
@@ -290,6 +290,27 @@ You can pass additional arguments to ``darker`` in the ``blackArgs`` option
 (e.g. ``["--diff", "--isort"]``), but make sure at least ``--diff`` is included.
 
 
+Vim
+---
+
+Unlike black and many other formatter darker needs access to the git history,
+therefore it does not work properly with the classical auto reformat plugins.
+
+You can though ask vim to run doarker on file save with the following in your
```
Comment (akaihola) [inline README.rst:299]:
```suggestion
You can though ask vim to run ``darker`` on file save with the following in your
```
typo

File: README.rst [hunk 4]
```diff
@@ -290,6 +290,27 @@ You can pass additional arguments to ``darker`` in the ``blackArgs`` option
...
+`.vimrc`:
+
+```
+autocmd BufWritePost *.py silent :!darker %
+```
+
+  - `BufWritePost` to run darker _once the file has been saved_,
+  - `silent` to not ask for confirmation each time,
+  - `:!` to run an external command,
+  - `%` for current file name.
```
Comment (akaihola) [inline README.rst:309]: The formatting was a bit off, this renders better in RestructuredText:

```suggestion
.. code-block:: vim

   autocmd BufWritePost *.py silent :!darker %

- ``BufWritePost`` to run darker *once the file has been saved*,
- ``silent`` to not ask for confirmation each time,
- ``:!`` to run an external command,
- ``%`` for current file name.
```

**Review by akaihola — CHANGES_REQUESTED** [review]
It's great to have detailed instructions for different IDE/editor environments! Thanks!

**Comment by Carreau** [comment]
Sorry, my brain was in markdown mode when I wrote this.

**Review by akaihola — APPROVED** [review]

**Comment by akaihola** [comment]
The build is failing due to the Black 20.8 update. Let's merge #58 first and rebase this one then.

**Comment by akaihola** [comment]
> The build is failing due to the Black 20.8 update. Let's merge #58 first and rebase this one then.

I did the rebase and had to open a new pull request #70 since the branch is now in my repository instead of @Carreau's.

**Pull request closed without merging**